iocage/lib/Jail.py

```diff
@@ -0,0 +1,500 @@
+            f"exec.timeout={self.config.exec_timeout}",
+            f"stop.timeout={self.config.stop_timeout}",
+            f"mount.fstab={self.path}/fstab",
+            f"mount.devfs={self.config.mount_devfs}"
+        ]
+
+        if self.host.userland_version > 9.3:
+            command += [
+                f"mount.fdescfs={self.config.mount_fdescfs}",
+                f"allow.mount.tmpfs={self.config.allow_mount_tmpfs}"
+            ]
+
+        command += [
+            "allow.dying",
+            f"exec.consolelog={self.logfile_path}",
+            "persist"
+        ]
+
+        humanreadable_name = self.humanreadable_name
+        try:
+            helpers.exec(command, logger=self.logger)
+            self.update_jail_state()
+            self.logger.verbose(
+                f"Jail '{humanreadable_name}' started with JID {self.jid}",
+                jail=self
+            )
+        except subprocess.CalledProcessError as exc:
+            code = exc.returncode
+            self.logger.error(
+                f"Jail '{humanreadable_name}' failed with exit code {code}",
+                jail=self
+            )
+            self.logger.verbose(exc.output, jail=self)
+            raise
+
+    def start_vimage_network(self):
+
+        self.logger.log("Starting VNET/VIMAGE", jail=self)
+
+        nics = self.config.interfaces
+        for nic in nics:
+
+            bridges = list(self.config.interfaces[nic])
+
+            try:
+                ipv4_addresses = self.config.ip4_addr[nic]
+            except:
+                ipv4_addresses = []
+
+            try:
+                ipv6_addresses = self.config.ip6_addr[nic]
+            except:
+                ipv6_addresses = []
+
+            net = Network.Network(
+                jail=self,
+                nic=nic,
+                ipv4_addresses=ipv4_addresses,
+                ipv6_addresses=ipv6_addresses,
+                bridges=bridges,
+                logger=self.logger
+            )
+            net.setup()
+            self.networks.append(net)
+
+    def stop_vimage_network(self):
+        for network in self.networks:
+            network.teardown()
+            self.networks.remove(network)
+
+    def set_nameserver(self):
+        self.config.resolver.apply(self)
+
+    def set_routes(self):
+
+        defaultrouter = self.config.defaultrouter
+        defaultrouter6 = self.config.defaultrouter6
+
+        if not defaultrouter or defaultrouter6:
+            self.logger.spam("no static routes configured")
+            return
+
+        if defaultrouter:
+            self.logger.verbose(
+                f"setting default IPv4 gateway to {defaultrouter}",
+                jail=self
+            )
+            self._set_route(defaultrouter)
+
+        if defaultrouter6:
+            self._set_route(defaultrouter6, ipv6=True)
+
+    def _set_route(self, gateway, ipv6=False):
+
+        ip_version = 4 + 2 * (ipv6 is True)
```

Comment:
Thanks. If you come up with a more obscure way to calculate the IP version, I'm happy to adapt 😼